cmake/Utils.cmake

```diff
@@ -278,3 +278,84 @@ function(parse_cuda_version CUDA_VERSION CUDA_VERSION_MAJOR_VAR CUDA_VERSION_MIN
+      VERBATIM )
+    add_library(${PYTHON_LIB_ARG_TARGET_NAME}_public INTERFACE)
+    add_library(${PYTHON_LIB_ARG_TARGET_NAME}_private INTERFACE)
+    target_sources(${PYTHON_LIB_ARG_TARGET_NAME}_private INTERFACE "${CMAKE_CURRENT_BINARY_DIR}/${PYTHON_LIB_ARG_TARGET_NAME}_dummy.cc")
+    target_sources(${PYTHON_LIB_ARG_TARGET_NAME}_private INTERFACE "${CMAKE_CURRENT_BINARY_DIR}/${PYTHON_LIB_ARG_TARGET_NAME}_dummy.cc")
+
+    foreach(PYVER ${PYTHON_VERSIONS})
+
+        set(PYTHON_LIB_TARGET_FOR_PYVER "${PYTHON_LIB_ARG_TARGET_NAME}_${PYVER}")
+        # check if listed python versions are accesible
+        execute_process(
+            COMMAND python${PYVER} -c "print('test', end='')"
+            RESULT_VARIABLE  PYTHON_EXISTS)
+
+        if (${PYTHON_EXISTS} EQUAL 0)
+            execute_process(
+                COMMAND python${PYVER}-config --extension-suffix
+                OUTPUT_VARIABLE PYTHON_SUFIX)
+            # remove newline and the extension
+            string(REPLACE ".so\n" "" PYTHON_SUFIX ${PYTHON_SUFIX})
+
+            execute_process(
+                COMMAND python${PYVER}-config --includes
+                OUTPUT_VARIABLE PYTHON_INCLUDES)
+            # split and make it a list
+            string(REPLACE "-I" "" PYTHON_INCLUDES ${PYTHON_INCLUDES})
+            string(REPLACE "\n" "" PYTHON_INCLUDES ${PYTHON_INCLUDES})
+            separate_arguments(PYTHON_INCLUDES)
+
+            add_library(${PYTHON_LIB_TARGET_FOR_PYVER} SHARED ${PYTHON_LIB_ARG_SRC})
+            target_link_libraries(${PYTHON_LIB_TARGET_FOR_PYVER} PUBLIC ${PYTHON_LIB_ARG_PUBLIC_LIBS})
+            target_link_libraries(${PYTHON_LIB_TARGET_FOR_PYVER} PRIVATE ${PYTHON_LIB_ARG_PRIV_LIBS})
+            target_link_libraries(${PYTHON_LIB_TARGET_FOR_PYVER} PRIVATE "-Wl,--exclude-libs,${PYTHON_LIB_ARG_EXCLUDE_LIBS}")
+
+            set_target_properties(${PYTHON_LIB_TARGET_FOR_PYVER}
+                                    PROPERTIES
+                                    LIBRARY_OUTPUT_DIRECTORY ${PYTHON_LIB_ARG_OUTPUT_DIR}
+                                    PREFIX "${PYTHON_LIB_ARG_PREFIX}"
+                                    OUTPUT_NAME ${PYTHON_LIB_ARG_OUTPUT_NAME}${PYTHON_SUFIX})
+            target_include_directories(${PYTHON_LIB_TARGET_FOR_PYVER}
+                                       PRIVATE "${PYBIND11_INCLUDE_DIR}"
+                                       PRIVATE "${pybind11_INCLUDE_DIR}")
+            # add includes
+            foreach(incl_dir ${PYTHON_INCLUDES})
+                target_include_directories(${PYTHON_LIB_TARGET_FOR_PYVER} PRIVATE ${incl_dir})
+            endforeach(incl_dir)
+
+            # add interface dummy lib as a dependnecy to easilly propagate options we could set from the above
+            target_link_libraries(${PYTHON_LIB_TARGET_FOR_PYVER} PUBLIC ${PYTHON_LIB_ARG_TARGET_NAME}_public)
+            target_link_libraries(${PYTHON_LIB_TARGET_FOR_PYVER} PRIVATE ${PYTHON_LIB_ARG_TARGET_NAME}_private)
```

Comment:
If you put sources in the interface library, this will populate the sources and there should be no need for the dummy.cc thing.

Reply:
Done

Reply:
Done